geth/common/types.go

```diff
@@ -173,28 +173,11 @@ type TxQueueManager interface {
 	DiscardTransactions(ids string) map[string]RawDiscardTransactionResult
 }
 
-// JailExecutor defines an interface which exposes method to be executed
-// against a Jail vm.
-type JailExecutor interface {
-	// Run exist so we are able to execute js code on pure otto.VM without runing
-	// it on the event loop.
-	Run(string) (otto.Value, error)
-
-	// Exec exists for the purpose to execute has normal on the event loop provided by
-	// ottoext.
-	Exec(string) (otto.Value, error)
-
-	// Fetch calls the underlying FetchAPI which makes http request
-	// to desired path. (See https://developer.mozilla.org/en/docs/Web/API/Fetch_API).
-	Fetch(string, func(otto.Value)) (otto.Value, error)
-}
-
 // JailCell represents single jail cell, which is basically a JavaScript VM.
 type JailCell interface {
-	CellVM() *otto.Otto
-	CellLoop() *loop.Loop
-	Executor() JailExecutor
-	Copy() (JailCell, error)
+	VM() *otto.Otto
```

Comment:
To remove `JailCell` interface, we will need to sure that `JailManager.NewJailCell` is not executed anywhere but from internal logic, else remove `NewJailCell` from `JailManager` interface.